framework/src/modules/random/stores/used_hash_onions.ts

```diff
@@ -90,4 +56,16 @@ export class UsedHashOnionsStore extends BaseOffchainStore<UsedHashOnionStoreObj
 			throw error;
 		}
 	}
+
+	public async setLatest(
```

Comment:
Could you please confirm or correct the implementation for `setLatest`, is it supposed to be simple update?

Reply:
I think how it's currently done is bit more complex. the code is around [here](https://github.com/LiskHQ/lisk-sdk/blob/18d9320b5d0179f258b94f454a228f4f134a2546/framework/src/modules/random/module.ts#L112-L137).

Assuming we have correct `nextHashOnion`, it's checking if the usedHashOnion with the same count exist in the list, overwrite otherwise push to the end.
This is because in case of reverting, it's possible that the used hash onion was also reverted; thus, it needs to use the same hash onion.